### PR TITLE
More fluid semester plot.

### PR DESCRIPTION
**Pull request by rpiaggio**
- Move semester plot computations to worker.
- Avoid restarting computations on resize.
- Correctly resize tile when shrinking height.
- Fix issue on WorkerServer where cancellation messages were not processed if there was a heavy ongoing computation.

### CODE REVIEW

File: common/src/main/webapp/less/style.less [hunk 1]
```diff
@@ -944,6 +944,7 @@ tfoot {
 .elevation-plot {
   flex-shrink: 1;
   flex-grow: 1;
+  height: 100%;
```
Comment (cquiroz) [inline common/src/main/webapp/less/style.less:947]: heh, that was the fix 😃

File: explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala [hunk 2]
```diff
@@ -119,7 +122,7 @@ object ElevationPlotSection {
       // Actual date
       .useState(ZonedDateTime.now(Site.GS.timezone).toLocalDate.plusDays(1))
       .render { (props, _, options, date) =>
-        implicit val ctx = props.ctx
+        import props.given
```
Comment (cquiroz) [inline explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala:125]: Nice

File: workers/src/main/scala/workers/PlotServer.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalTime
+import scala.ContextFunction1
+import scala.collection.immutable.TreeMap
+import scala.scalajs.js.annotation.JSExport
+import scala.scalajs.js.annotation.JSExportTopLevel
+
+@JSExportTopLevel("PlotServer", moduleID = "exploreworkers")
+object PlotServer extends WorkerServer[IO, PlotMessage.Request] with IOApp.Simple {
```
Comment (cquiroz) [inline workers/src/main/scala/workers/PlotServer.scala:35]: Are we using `IOApp`? What about reloads?

**Review by cquiroz — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/tabs/ElevationPlotTile.scala [hunk 4]
```diff
@@ -30,7 +31,8 @@ object ElevationPlotTile {
       "Elevation Plot",
       canMinimize = true,
       bodyClass = ExploreStyles.ElevationPlotTileBody.some,
-      tileClass = ExploreStyles.ElevationPlotTile.some
+      tileClass = ExploreStyles.ElevationPlotTile.some,
+      sizeStateCallback = s => Callback.log(s)
```
Comment (toddburnside) [inline explore/src/main/scala/explore/tabs/ElevationPlotTile.scala:35]: Do you want to keep this logging here?

File: explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala [hunk 5]
```diff
@@ -43,17 +43,17 @@ import react.semanticui.elements.button.ButtonGroup
...
-  uid:              User.Id,
-  tid:              Target.Id,
-  scienceMode:      Option[ScienceMode],
-  coords:           Coordinates
-)(implicit val ctx: AppContextIO)
+  uid:           User.Id,
+  tid:           Target.Id,
+  scienceMode:   Option[ScienceMode],
+  coords:        Coordinates
+)(using val ctx: AppContextIO)
```
Comment (toddburnside) [inline explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala:50]: With the `import props.given` below, does this need a name anymore?

**Review by toddburnside — APPROVED** [review]
LGTM

File: workers/src/main/scala/workers/PlotServer.scala (hunk 3 above)
Comment (rpiaggio) [reply workers/src/main/scala/workers/PlotServer.scala:35]: Good catch, that was just a test, let me revert.

**Comment by rpiaggio** [comment]
> The semester plot used to have a line indicating progress, which would move along the calculation. Do we still have that? Is the worker sending messages along to inform progress to the UI?

Yes, that's still there. The worker sends each value as they become available, the UI sees a stream of values and adds each to the plot as soon as they are received, advancing the progress line.

File: explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala (hunk 5 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala:50]: Unfortunately, it does: we need it as a `val`, which forces us to declare a name.

Comment (toddburnside) [reply explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala:50]: Ah, yeah. Second parameter list is private by default.

**Pull request merged**